Formula/libtensorflow.rb

```diff
@@ -20,11 +20,15 @@ class Libtensorflow < Formula
   end
 
   def install
-    # Allow tensorflow to use current version of bazel
-    (buildpath / ".bazelversion").atomic_write Formula["bazel"].version
-
+    optflag = if Hardware::CPU.arm? && OS.mac?
+      "-mcpu=apple-m1"
+    elsif build.bottle?
+      "-march=#{Hardware.oldest_cpu}"
+    else
+      "-march=native"
+    end
```

Comment:
Honestly, reading from `ENV["HOMEBREW_OPTFLAGS"]` should be enough here.

It currently returns nothing for arm64, but that's because our CI is the oldest hardware anyway so I haven't actually put in the effort to add it (it's a little tricky when dealing with different compilers).

Reply:
The build hangs if nothing is provided, so there will still need arm64-dedicated logic. 

If that returns multiple values, may need to add extra logic when passing values to Bazel `--copt`. Not sure if the flag allows you to pass in multiple values at once. Documentation example shows:
```
  % bazel build --copt="-g0" --copt="-fpic" //foo
```
So may need some split/map logic.

Reply:
It never currently provides multiple args, but I wouldn't rely on that remaining the case.

But I'm confused what we need to pass `--copt` anyway - is what `configure` does to set that up insufficient?

> The build hangs if nothing is provided, so there will still need arm64-dedicated logic

Yeah seems like their configure script is sorely missing an unattended mode.

```rb
ENV["HOMEBREW_OPTFLAGS"].presence || " "
```

This will however cover it though. You can still have if an arm64 case if you want but I'd still personally handle empty strings.

Reply:
> But I'm confused what we need to pass `--copt` anyway - is what `configure` does to set that up insufficient?

Not sure. Looks like it has been around since the beginning #10273. Haven't checked myself.

Configure does update a bazelrc file https://github.com/tensorflow/tensorflow/blob/v2.8.0/configure.py#L541-L543
```py
  for opt in cc_opt_flags.split():
    write_to_bazelrc('build:opt --copt=%s' % opt)
    write_to_bazelrc('build:opt --host_copt=%s' % opt)
```

Probably safe to simplify the build and remove the `--copt`.

Reply:
One thing I notice is that `-march=native` can be passed into `CC_OPT_FLAGS` and it doesn't impact the ARM build unless it is passed in as `--copt=-march=native` (even though it is not compatible with clang).

Reply:
So, how are we going to handle it?